Libraries/Opc.Ua.Client/Session/Session.cs

```diff
@@ -3652,6 +3745,279 @@ public bool RemoveTransferredSubscription(Subscription subscription)
+                    resultForPass.AddRange( referenceDescriptionsForNextPass );
+                    referenceDescriptionsForNextPass.Clear();
+
+                    errorsForPass = new List<ServiceResult>();
+                    errorsForPass.AddRange( errorsForNextPass );
+                    errorsForNextPass.Clear();
+
+                    nodesToBrowseForPass = new List<NodeId>();
+                    nodesToBrowseForPass.AddRange( nodesToBrowseForNextPass );
+                    nodesToBrowseForNextPass.Clear();
+
+                    String aggregatedErrorMessage = "ManagedBrowse: in pass {0}, {1} {2} occured with a status code {3}";
+
+                    if (badCPInvalidErrorsPerPass > 0)
+                    {                         
+                        Utils.LogInfo(aggregatedErrorMessage, passCount, badCPInvalidErrorsPerPass,
+                            badCPInvalidErrorsPerPass == 1 ? "error" : "errors", "BadContinuationPointInvalid.");
+                    }
+                    if( badNoCPErrorsPerPass > 0 )
+                    {
+                        Utils.LogInfo(aggregatedErrorMessage, passCount, badNoCPErrorsPerPass,
+                            badNoCPErrorsPerPass == 1 ? "error" : "errors", "BadNoContinuationPoints.");
+                    }
+                    if( otherErrorsPerPass > 0)
+                    {
+                        Utils.LogInfo(aggregatedErrorMessage, passCount, otherErrorsPerPass,
+                            otherErrorsPerPass == 1 ? "error" : "errors", "different from BadNoContinuationPoints or BadContinuationPointInvalid.");
+                    }
+                    if (otherErrorsPerPass == 0 && badCPInvalidErrorsPerPass == 0 && badNoCPErrorsPerPass == 0)
+                    {
+                        Utils.LogTrace("MangedBrowse completed with no errors.");
+                    }
+
+                    passCount++;
+
+                } while ( nodesToBrowseForPass.Count > 0 );                
+            }
+            catch (Exception ex)
+            {
+                Utils.LogError(ex, "ManagedBrowse failed");
+            }
+        }
+
+        private class ReferenceWrapper<T>
+        {
+            public T reference { get; set; }
+        }
+
+        /// <summary>
+        /// Call browse and, if necessary, browse next (recursively) on input
+        /// Release continuation points on all errors other than
+        /// BadNoContinuationPoints and BadContinuationPointInvalid  
+        /// </summary>
+        /// <param name="view"></param>
+        /// <param name="nodeIds"></param>
+        /// <param name="maxResultsToReturn"></param>
+        /// <param name="browseDirection"></param>
+        /// <param name="referenceTypeId"></param>
+        /// <param name="includeSubtypes"></param>
+        /// <param name="nodeClassMask"></param>
+        /// <param name="result"></param>
+        /// <param name="finalErrors"></param>
+        /// <returns></returns>
```

Comment:
if there is no description for the parameters, rather remove them

Reply:
I believe I fixed most of these by now.